src/runtime/policy/policy.ts

```diff
@@ -232,6 +233,16 @@ export class PolicyTarget {
     });
   }
 
+  getRestrictedFields() {
+    return this.fields.map(f => f.getRestrictedFields(this.type))
+          .reduce((fields, field) => ({...fields, [field[0]]: field[1]}), {});
```

Comment:
Could you return an object from `getRestrictedFields` rather than an array? Then use destructuring to make it easier to read here:

```
.reduce((fields, {name, restrictedField}) => ({...fields, [name]: restrictedField}))
```